postcommit.py

```diff
@@ -0,0 +1,642 @@
+import mercurial, urllib, subprocess
+from mercurial import util
+from mercurial import patch
+from mercurial.ui import ui
+import json, os, inspect, sys, urllib2, httplib, datetime, re
+
+from mercurial.localrepo import localrepository
+
+
+repopath = sys.argv[1]
+type = sys.argv[2]
+if len(sys.argv) > 4:
+	oldId = sys.argv[3]
+	newId = sys.argv[4]
+	branch = sys.argv[5]
+elif len(sys.argv) > 3:
+	st_changeset = sys.argv[3]
+else:
+	st_changeset = ""
+
+
+COMMIT_TEMPLATE = 'LEGAL DECLARATION:I hereby ([^\n])*\n(LICENSE COMPATIBLE:([Yy][Ee][Ss]|[Nn][Oo]))*\n*'
+OPTIONAL_TEMPLATE = 'IGNORE_ISSUECHECKS:YES(\n)*(I18N ISSUE ID :)*'
+
+def getFileType(filename):
+    values = filename.split('.')
+    if len(values) > 1:
+        return values[-1]
+    else:
+        return 'others'
+
+def escapeChars(string):
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename function "escapeChars" to match the regular expression ^[a-z_][a-z0-9_]{2,}$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS1542)